### ISSUE TITLE
DWLS requires an update

### ISSUE THREAD

**Issue by LorenzoMerotto** [omnideconv/omnideconv#119]
Hello everyone, 

Me and @alex-d13 noticed that DWLS currently gives an error when Seurat is used as the method to create the signature matrix. This one could help us dealing with big single cell datasets. 

The problem is on the method's side, so to solve it we should either contact the authors or make a patch. I think I could (easily?) patch it, but the version we currently used is stored in a fork that Konstantin made from the original repository. 

I think it could be useful to just fork all the repositories of the patched methods inside omnideconv, so htat if we have to make some edits it is easier for everybody. 

What do you think about it? @grst @mlist @federicomarini @PelzKo

**Comment by grst** [comment]
Yeah, I think we need more control over all those packages that are not maintained. Having them in the omnideconv namespace is definitely better than having them in @PelzKo's or mine. We should just make clear in the README that the methods are a fork, not actively maintained by us (except keeping them running) and not "official" omnideconv repos.

Another option would be to copy the source code into omnideconv, but that can quickly become a mess, and challenging with licenses (e.g. technically you can't have GPL2 and GPL3 code in the same repo, not that anyone cared :shrug:)

**Comment by mlist** [comment]
I agree with Gregor here but would only fork on a per need basis, i.e. once we need to patch something.

**Comment by grst** [comment]
Sure!
We already have four (?) forked repos already now under my or Konstantin's account which we could get in.

@LorenzoMerotto, would you take care of that?

**Comment by LorenzoMerotto** [comment]
I could take care of it no problem but I think since you have admin rights on your repository you have to do it, right? Same as we did for omnideconv in the first place.

**Comment by grst** [comment]
you can just fork mine & konstantin's repos again.

**Comment by LorenzoMerotto** [comment]
These are the dependencies listed on the omnideconv DESCRIPTION file: 
```
grst/MOMF,
grst/SCDC,
PelzKo/CDSeq_R_Package,
PelzKo/dwls,
```

However @grst I see that among your repositories you have CDseq, DWLS, Bisque and BisqueRNA as well. Which ones should I consider?

**Comment by grst** [comment]
only those from the description.
I think the rest was also patched, but the changes were merged back into the original repos.

**Comment by alex-d13** [comment]
I have some doubts with this Seurat implementation, especially concerning the default parameters to select the cell type marker genes. I tried it out using the example dataset in omnideconv, and got not a single significant gene for some cell types. Then DWLS also cannot build the signature matrix correctly and crashes.
I feel like the default pvalue and FC cutoffs are for the MAST implementation, which is the default run mode:
> Upon characterization of the cell types, differential expression analysis is performed to identify marker genes for each cell type. We define marker genes as genes with an FDR adjusted p-value of <0.01 (defined using the hurdle model in the MAST R package), and a log2 mean fold change >0.5. For very large single-cell data sets like the Mouse Cell Atlas, p-values are instead determined using the Seurat R package under the bimod likelihood ratio test for single-cell gene expression[31](https://www.nature.com/articles/s41467-019-10802-z#ref-CR31), due to the faster runtime.

Would you say we just use the parameters for Seurat either way and if it does not produce a signature for one benchmakring combination, we just skip the DWLS+Seurat results for this combination? Or just stick with the MAST (default) DWLS run?

**Comment by LorenzoMerotto** [comment]
@alex-d13 For once I have actually a good news: no gene is found for the signature matrix with the current Seurat implementation and parameters just because DWLS was based on an old Seurat version and the variable names used in the process are deprecated. I fixed this on Friday evening, that's why I did not notice anyone. The updated version is in a branch in the dwls repository in omnideconv, [here](https://github.com/omnideconv/dwls/commit/49b5682d67c2522aa9bdfe0e10b1cff63b10cb65). 
I tested it and it works, menaing that some genes are selected. We can still question the choice of parameters
